File: packages/js-dpp/lib/blockTimeWindow/ValidationResult.js
class ValidationResult {
  /**
   *
   * @param {boolean} isValid
   * @param {Date} timeWindowStart
   * @param {Date} timeWindowEnd
   */
  constructor(isValid, timeWindowStart, timeWindowEnd) {
    this.valid = isValid;
    this.timeWindowStart = timeWindowStart;
    this.timeWindowEnd = timeWindowEnd;
  }

  /**
   *
   * @returns {Date}
   */
  getTimeWindowStart() {
    return this.timeWindowStart;
  }

  /**
   *
   * @returns {Date}
   */
  getTimeWindowEnd() {
    return this.timeWindowEnd;
  }

  /**
   *
   * @returns {boolean}
   */
  isValid() {
    return this.valid;
  }
}

module.exports = ValidationResult;

File: packages/rs-dpp/src/block_time_window/validation_result.rs
pub struct TimeWindowValidationResult {
    pub valid: bool,
    pub time_window_start: u64,
    pub time_window_end: u64,
}

impl TimeWindowValidationResult {
    pub fn get_time_window_start(&self) -> u64 {
        self.time_window_start
    }

    pub fn get_time_window_end(&self) -> u64 {
        self.time_window_end
    }

    pub fn is_valid(&self) -> bool {
        self.valid
    }
}
